lib/spack/spack/compilers/aocc.py

```diff
@@ -118,3 +119,21 @@ def f77_version(cls, f77):
     @property
     def stdcxx_libs(self):
         return ('-lstdc++', )
+
+    @property
+    def cflags(self):
+        return self._handle_default_flag_addtions()
+
+    @property
+    def cxxflags(self):
+        return self._handle_default_flag_addtions()
+
+    @property
+    def fflags(self):
+        return self._handle_default_flag_addtions()
+
+    def _handle_default_flag_addtions(self):
+        # This is a known issue for AOCC 3.0 see:
+        # https://developer.amd.com/wp-content/resources/AOCC-3.0-Install-Guide.pdf
+        if self.real_version == ver('3.0.0'):
+            return "-Wno-unused-command-line-argument -mllvm -eliminate-similar-expr=false"
```

Comment:
Feel free to introduce a new variable to solve these issue. I think the following may also work:
```suggestion
            return ("-Wno-unused-command-line-argument "
                    "-mllvm -eliminate-similar-expr=false")
```